lib/Doctrine/DBAL/Connection.php

```diff
@@ -1297,6 +1305,11 @@ public function commit()
             }
         }
 
+        $this->commitDone();
+    }
+
+    private function commitDone()
```

Comment:
` : void`

Also: naming suggests that committing finished, yet the code below begins a new transaction now?

Reply:
Documentation block also needed - or better/more explicative naming.

Reply:
Oh yeah, I wasn't sure about that part at all :D Do you have any suggestions for the name? It basically just resets the internal state of the object, after the commit.

Reply:
Purpose of this method is to set the `Connection` to the correct state. Until now the code didn't expect the `commit()` to ever throw any exception. Now that it is possible we need to manage the state so that if someone catches the exception, then they can still use the connection afterwards. Without doing that the connection would be stuck with `transactionNestingLevel` set to 1 even though the transaction is long gone. The `->beginTransaction()` is there because of the `autoCommit` thingie - without it the `Connection` object would again be in an unexpected state.

Reply:
Right, and that isn't clear by reading the method: can we somehow improve that?

Consider comments to be the last resort: having a clear method name would be a good improvement at first.

Reply:
@Ocramius added the patch. Does `updateTransactionStateAfterCommit()` make more sense?